install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -274,12 +274,12 @@ data:
 
 
 
-{{- if hasKey .Values.bpf "mapDynamicSizeRatio" }}
-  # Specifies the ratio (0.0-1.0) of total system memory to use for dynamic
+{{- if .Values.bpf.mapDynamicSizeRatio }}
+  # Specifies the ratio (0.0-1.0] of total system memory to use for dynamic
   # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
   bpf-map-dynamic-size-ratio: {{ .Values.bpf.mapDynamicSizeRatio | quote }}
 {{- else if ne $defaultBpfMapDynamicSizeRatio 0.0 }}
-  # Specifies the ratio (0.0-1.0) of total system memory to use for dynamic
+  # Specifies the ratio (0.0-1.0] of total system memory to use for dynamic
   # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
   bpf-map-dynamic-size-ratio: {{ $defaultBpfMapDynamicSizeRatio | quote }}
```

Comment:
How can detect a cluster that's performing an upgrade for which it never explicitly set `mapDynamicSizeRatio` in its values? My understanding is that if we are setting a default of 0.0025 that means clusters that never set `mapDynamicSizeRatio` will default to 0.0025 which will resize bpf maps.

Reply:
so I did a quick test to install cilium 1.11.0 `cilium install --version=v1.11.0 --kube-proxy-replacement strict --helm-set-string=k8sServiceHost=10.3.72.239,k8sServicePort=6443,ipam.operator.clusterPoolIPv4PodCIDR="10.42.0.0/16"`, you can see I did not specify helm option `bpf.mapDynamicSizeRatio`, but the cilium agent has `level=info msg="  --bpf-map-dynamic-size-ratio='0.0025'" subsys=daemon`, so the default `mapDynamicSizeRatio` is set to 0.0025 since version `>=1.8`, this PR change does not change that behavior, am I missing anything?

Reply:
@aanm note that `$defaultBpfMapDynamicSizeRatio` is untouched by this PR, my understanding is that a >=v1.8 cluster without explicit configuration of `mapDynamicSizeRatio` already default to `0.0025`, see [here](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/install/kubernetes/cilium/templates/cilium-configmap.yaml#L21).

In fact, the [CLI default value of `0.0`](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/daemon/cmd/daemon_main.go#L815) is not in the accepted range (see [here](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/pkg/option/config.go#L3503-L3518)) but that's arguably out of the scope of this PR.

Reply:
@kaworu @vincentmli could we compromise and leave the changes on this PR but making them in a different commit, what do you think?

Reply:
Thank you for taking a look. LGTM